test/Microsoft.Health.Extensions.Fhir.R4.UnitTests/FhirServiceExceptionProcessorTests.cs

```diff
@@ -0,0 +1,102 @@
+                new object[] { _httpNotKnownEx, "FHIRServiceErrorConfigurationError", nameof(ErrorSource.User) },
+                new object[] { _httpEx, "FHIRServiceErrorHttpRequestError" },
+                new object[] { _msalInvalidResourceEx, "FHIRServiceErrorConfigurationError", nameof(ErrorSource.User) },
+                new object[] { _msalInvalidScopeEx, "FHIRServiceErrorConfigurationError", nameof(ErrorSource.User) },
+                new object[] { _msalEx, "FHIRServiceErrorMsalServiceErrortest_code" },
+                new object[] { _ex, "FHIRServiceErrorGeneralError" },
+            };
+
+        public static IEnumerable<object[]> CustomizeExceptionData =>
+            new List<object[]>
+            {
+                new object[] { _fhirForbiddenEx, typeof(UnauthorizedAccessFhirServiceException) },
+                new object[] { _fhirNotFoundEx, typeof(InvalidFhirServiceException) },
+                new object[] { _fhirBadRequestEx, typeof(FhirOperationException) },
+                new object[] { _argEndpointNullEx, typeof(InvalidFhirServiceException) },
+                new object[] { _argEndpointEx, typeof(InvalidFhirServiceException) },
+                new object[] { _argEx, typeof(ArgumentException) },
+                new object[] { _uriEx, typeof(InvalidFhirServiceException) },
+                new object[] { _httpNotKnownEx, typeof(InvalidFhirServiceException) },
+                new object[] { _httpEx, typeof(HttpRequestException) },
+                new object[] { _msalInvalidResourceEx, typeof(InvalidFhirServiceException) },
+                new object[] { _msalInvalidScopeEx, typeof(InvalidFhirServiceException) },
+                new object[] { _msalEx, typeof(MsalServiceException) },
+                new object[] { _ex, typeof(Exception) },
+            };
+
+        [Theory]
+        [MemberData(nameof(ProcessExceptionData))]
+        public void GivenExceptionType_WhenProcessException_ThenExceptionLoggedAndErrorMetricLogged_Test(Exception ex, string expectedErrorMetricName, string expectedErrorSource = null)
+        {
+            var logger = Substitute.For<ITelemetryLogger>();
+
+            FhirServiceExceptionProcessor.ProcessException(ex, logger);
+
+            logger.ReceivedWithAnyArgs(1).LogError(ex);
+            logger.Received(1).LogMetric(
+                Arg.Is<Metric>(m =>
+                    m.Name.Equals(expectedErrorMetricName) &&
+                    m.Dimensions[DimensionNames.Name].Equals(expectedErrorMetricName) &&
+                    m.Dimensions[DimensionNames.Operation].Equals(ConnectorOperation.FHIRConversion) &&
+                    m.Dimensions[DimensionNames.Category].Equals(Category.Errors) &&
+                    m.Dimensions[DimensionNames.ErrorType].Equals(ErrorType.FHIRServiceError) &&
+                    m.Dimensions[DimensionNames.ErrorSeverity].Equals(ErrorSeverity.Critical) &&
```

Comment:
nit: Maybe this can go into a separate method like ValidateFHIRServiceErrorProperteies() similar to a recent change you made in the other test file.